### PR TITLE
Update Query Filtering to be based on acls again

### PR DESCRIPTION
**Pull request by ryanrath**
## Description
This PR's purpose is to replace the hard coded logic of
`Parameters::getParameters`, the function that determines how a query should be
filtered based on what a users most privileged acl is, with a database query
that utilizes the `user_acl_group_by_parameters` table.

To accomplish this we first need to capture for each acl which, if any,
dimension filtering is required. Thankfully our configuration files contain this
information already in the `dimensions` property. A new table was created
`acl_dimensions` that serve as a relation table between `acls` and `group_bys` (
dimension is another name for a group_by, and seeing as we already have an
`acl_group_bys` table, `acl_dimensions` it is. ). To ensure this table is
populated correctly the `acl-config` script was updated to include this as
another managed table. Next, to ensure that when a user is created or modified
the appropriate records are created in the `user_acl_group_by_parameeters` table
the `XDUser::saveUser` function was updated to include the maintenance of these
records.

New component tests ( and artifacts ) have been added to exercise the
`Parameters::getParameters` function.

## Motivation and Context
The old function, while consolidating the logic of which filters should be applied in which cases, did so via hard coded `logic` which is not only not ideal, but we have a table ( `user_acl_group_by_parameters`) that is designed to provide this logic. It is also in keeping with our established design tenant of exclusively using the database at runtime versus configuration files which the old function was also not doing. 

## Tests performed
All automated test suites executed & passed for both the upgrade and install paths. 

## Checklist:

- [X] The pull request description is suitable for a Changelog entry
- [X] The milestone is set correctly on the pull request
- [X] The appropriate labels have been added to the pull request

### CODE REVIEW

File: classes/Models/Services/Parameters.php [hunk 1]
```diff
@@ -2,8 +2,15 @@
 
 namespace Models\Services;
 
-use Configuration\XdmodConfiguration;
+use CCR\DB;
 
+/**
+ * Class Parameters
+ *
+ * This class is meant to provide the
```
Comment (jtpalmer) [inline classes/Models/Services/Parameters.php:10]: Provide the?

**Review by jtpalmer — CHANGES_REQUESTED** [review]
Looks fine as far as I can tell.  Not sure why the upgrade is failing.  I just have some requests for doc updates.

File: bin/acl-config [hunk 2]
```diff
@@ -875,7 +877,7 @@ function processAcls(iDatabase $db, $module, array $realms, array $acls, array $
  * @return int|null null if the acl was not able to be created or the acl_id if
  *                       it was created or exists.
  **/
-function processAcl(iDatabase $db, $module, $name, $display, $type = null, $hierarchies = null)
+function processAcl(iDatabase $db, $module, $name, $display, $type = null, $hierarchies = null, $dimensions = null)
```
[the remark this answers is not in the document]
Comment (ryanrath) [reply bin/acl-config:882]: Sorry about missing your review on this PR, all of your requested changes should be in the latest commit.

File: bin/acl-config [hunk 3]
```diff
@@ -1152,6 +1159,52 @@ SQL;
     $log->notice("Processed: $processed, Inserted: $inserted");
 }
 
+/**
+ * @param iDatabase $db
+ * @param string    $aclName
+ * @param array     $dimensions
```
[the remark this answers is not in the document]
Comment (ryanrath) [reply bin/acl-config:1172]: Definitely, I've updated the doc blocks in the latest commit.

File: classes/Models/Services/Parameters.php (hunk 1 above)
Comment (ryanrath) [reply classes/Models/Services/Parameters.php]: Answer to life the Universe and Everything? I wish... I have finished writing the documentation block for this class in the latest commit though.

**Review by jtpalmer — APPROVED** [review]

**Pull request merged**